apps/nextra/pages/en/network/blockchain/execution.mdx

```diff
@@ -0,0 +1,46 @@
+---
+title: "Execution"
+---
+
+# Execution
+
+In Aptos, execution refers to the process by which validators 
+run and apply smart contract transactions from ordered blocks to update the blockchain’s state.
+
+It is crucial because validators must agree on the same final state, post-execution,
+ensuring consistency and security across the network.
+
+Efficient execution, particularly parallel execution, is key to scaling blockchain performance and throughput.
+
+## Why Parallel Execution?
+
+![](https://miro.medium.com/v2/resize:fit:4800/format:webp/0*qG0KRe1KzsP8Odth)
+
+Smart contract execution is a major throughput bottleneck for blockchains. 
+After proposing blocks and agreeing on their order, validators have to execute the 
+transactions in the ordered blocks. Crucially, validators must arrive at the same final state, 
+which must correspond to some sequential execution of the transactions. 
+Due to the lack of a foundational solution, current blockchains either execute 
+sequentially or require an embarrassingly parallel workload (i.e. no inner conflicts) 
+for improved performance. Sequential execution does not scale well, and assuming that 
+all transactions are commutative is unrealistic when considering a broad range of smart contracts. 
+
+In fact, transactions in a blockchain can have a significant number of access conflicts due to 
+potential performance attacks, accessing popular contracts 
+(e.g. due to economic opportunities such as auctions and arbitrage).
+
+## Block-STM
+
+We designed and implemented a highly efficient, multi-threaded, in-memory parallel 
+execution engine. It leverages the preset order of transactions and combines 
+Software Transactional Memory techniques with a novel collaborative schedule.
+
+Block-STM is the state-of-the-art dynamic parallelism execution engine developed by the Aptos Labs team.
+
+Dynamic parallelism allows developers to flexibly write applications without 
+facing design constraints of statically declaring write sets of transactions.
+
```

Comment:
There's a lot of jargon thrown around, it reads very much like the papers.  This is hopefully a simplified description

```suggestion
We took a different approach.  Instead of forcing the burden on the developer to specify the conflicts between transactions, we wanted the execution engine to automatically handle them.  We created Block-STM, a multi-threaded parallel execution engine, that can execute over 160k transactions per second.  Block-STM leverages the preset order of transactions, using [STM](https://en.wikipedia.org/wiki/Software_transactional_memory) techniques used in traditional computing with a novel collaborative schedule.

Dynamic parallelism (also known as optimistic concurrency) will adjust to the smart contracts being run, and detect conflicts.  When there's a conflict it will rerun those transactions sequentially in the predetermined order to ensure that the final state would be the same as if the whole block was run sequentially.  This removes the burden from the developer to specify dependencies, and allows to automatically parallelize all transactions that do not conflict.
```